DeviceAdapters/Zaber/ConnectionManager.cpp

```diff
@@ -0,0 +1,39 @@
+#include "ConnectionManager.h"
+
+std::shared_ptr<zml::Connection> ConnectionManager::getConnection(std::string port, MM::Core* core)
+{	
+	std::lock_guard<std::mutex> lockGuard(lock_);
+	if (connections_.count(port)) {
+		if (auto connectionPtr = connections_.at(port).lock()) {
+			return connectionPtr;
+		}
+	}
+
+	auto connection = std::make_shared<zml::Connection>(zml::Connection::openSerialPort(port));
+	auto id = connection->getInterfaceId();
+	connection->getDisconnected().subscribe([=](std::shared_ptr<zmlbase::MotionLibException>) {
+		removeConnection(port, id);
+	});
+	connections_[port] = connection;
```

Comment:
Race condition - this should go before the subscribe.

Reply:
I don't see it.

Reply:
there is a lock in removeConnectio too.